src/sortStudents.ts

```diff
@@ -1,16 +1,97 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverageGrades(arr: number[]): number {
+  return arr.reduce((sum: number, curr: number) => sum + curr, 0) / arr.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return studentsCopy.sort((
+        studentA: Student,
+        studentB: Student,
+      ) => {
+        if (order === 'asc') {
+          return studentA[sortBy].localeCompare(studentB[sortBy]);
+        }
+
+        return studentB[sortBy].localeCompare(studentA[sortBy]);
+      });
+
+    case SortType.Surname:
+      return studentsCopy.sort((
+        studentA: Student,
+        studentB: Student,
+      ) => {
+        if (order === 'asc') {
+          return studentA[sortBy].localeCompare(studentB[sortBy]);
+        }
+
+        return studentB[sortBy].localeCompare(studentA[sortBy]);
+      });
+
+    case SortType.Age:
+      return studentsCopy.sort((
+        studentA: Student,
+        studentB: Student,
+      ) => {
+        if (order === 'asc') {
+          return studentA[sortBy] - studentB[sortBy];
+        }
+
+        return studentB[sortBy] - studentA[sortBy];
+      });
+
+    case SortType.Married:
+      return studentsCopy.sort((
+        studentA: Student,
+        studentB: Student,
+      ) => {
+        if (order === 'asc') {
+          return +studentA[sortBy] - +studentB[sortBy];
```

Comment:
I would use `Number` it's more understandable and noticable in the code
```suggestion
          return Number(studentA[sortBy]) - Number(studentB[sortBy]);
```